Network/HTTP/Auth.hs

```diff
@@ -36,6 +36,11 @@ import Data.Char
 import Data.Maybe
 import Data.Word ( Word8 )
 
+#include <ghcplatform.h>
+#ifdef HaLVM_TARGET_OS
+#define MIN_VERSION_network(a,b,c) 1
```

Comment:
So, it turns out that this isn't as clean as I'd hoped. In particular -- and now I'm remembering why I did this condition this way 😃  -- CPP chokes when it sees a CPP macro that it doesn't know. (Which is probably the right thing to do, even though I kind of which they did the wrong thing.)

What that means is that there are three options:

Option #1: Do it the way I sort of have here, where I create a fake `MIN_VERSION_network` to reference later when it's not defined. I could/should clear this up by using something other than `#ifdef HaLVM`; something like `#ifdef MIN_VERSION_network_hans` would be better, probably.

Option #2: I could create a neutral switch (`USE_MODERN_RELATIVETO`) and define that at the top of the file, based on the values of `MIN_VERSION_network_hans` or `MIN_VERSION_network` as appropriate. It'd be a little ugly, but at least it'd be outside of a code black. By ugly, I mean it's going to have to look something like:

```
#ifdef MIN_VERSION_network
#define USE_MODERN_RELATIVETO MIN_VERSION_netwok(2,4,0)
#endif
#ifdef MIN_VERSION_network_hans
#define USE_MODERN_RELATIVETO MIN_VERSION_network_hans(2,4,0)
#endif
```

Option #3: If you'd be willing to drop support for `network` < 2.4, then we could get rid of all the CPP and solve it that way.